### PR TITLE
Check Shadowing

### PR DESCRIPTION
**Pull request by Drapegnik**
solved #1

### CODE REVIEW

File: app/Main.hs [hunk 1]
```diff
@@ -58,5 +59,25 @@ eval env (App t u) =
             Just value -> eval ((arg, value) : env') expr
         _                            -> Nothing
 
+
+-- checkShadowing [] shadow_abs -> ["y","x"]
+-- checkShadowing [("x", Nill)] shadow_abs -> ["x","y","x"]
```
Comment (Drapegnik) [inline app/Main.hs]: @DaQuirm is it ok?

File: app/Main.hs [hunk 2]
```diff
@@ -40,7 +40,8 @@ test_err_1 =
   App (App l_id (Lit "*")) (Lit "!")
 
 data Value
-  = Value String
+  = Nill
```
Comment (DaQuirm) [inline app/Main.hs]: This constructor should not be necessary now

File: app/Main.hs [hunk 3]
```diff
@@ -58,5 +58,33 @@ eval env (App t u) =
             Just value -> eval ((arg, value) : env') expr
         _                            -> Nothing
 
+checkShadowing :: [Identifier] -> Expr -> [Identifier]
+checkShadowing args (Abs arg expr) 
+  | arg !! 0 == '_' = nested
```
Comment (DaQuirm) [inline app/Main.hs]: Why not just `arg == '_'`?
Otherwise we allow all identifiers beginning with `_`.
NB: `!!` is unsafe 😕

Comment (Drapegnik) [reply app/Main.hs]: this is not a bug but a feature 😄:

Comment (DaQuirm) [reply app/Main.hs]: okay, so be it. we'll remove it if it gets too confusing 👍

**Pull request merged**